controllers/overlaynetwork_controller.go

```diff
@@ -47,9 +54,25 @@ type OverlayNetworkReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *OverlayNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx).WithValues("overlaynetwork", req.NamespacedName)
+
+	logger.Info("Reconciling OverlayNetwork")
 
-	// TODO(user): your logic here
+	overlayNetwork := &selfservicev1.OverlayNetwork{}
+	if err := r.Client.Get(ctx, req.NamespacedName, overlayNetwork); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get OverlayNetwrok %q: %v", req.NamespacedName, err)
+	}
+
+	netAttachDef := renderNetAttachDef(overlayNetwork)
+	if err := r.Create(ctx, netAttachDef); err != nil {
+		if errors.IsAlreadyExists(err) {
+			logger.Info("NetworkAttachmentDefinition [%q] already exist", req.NamespacedName)
```

Comment:
The log is redundant, the error will be anyway printed to the log.

Reply:
Done